File: tests/lind-wasm-tests/memcpy/memcpy.c
#include <stdio.h>
#include <string.h>
#include <unistd.h>

int main() {
    // Source and destination buffers
    char src[] = "Hello, World!";
    char dst[50];

    // Use memcpy to copy data from src to dst
    memcpy(dst, src, strlen(src) + 1);  // +1 to include the null terminator

    // Use the write syscall to write the dst buffer to stdout
    write(1, dst, strlen(dst));

    // Return success
    return 0;
}

File: tests/unit-tests/memory_tests_rust/deterministic/memcpy.rs
use std::io::{self, Write};

fn main() -> io::Result<()> {
    // Source and destination buffers
    let src = "Hello, World!";
    let mut dst = [0u8; 50];
    
    // Use Rust's copy_from_slice to copy data from src to dst
    let src_bytes = src.as_bytes();
    dst[..src_bytes.len()].copy_from_slice(src_bytes);
    
    // Use stdout to write the dst buffer (equivalent to write syscall)
    io::stdout().write_all(&dst[..src_bytes.len()])?;
    io::stdout().flush()?;
    
    Ok(())
} 